### PR TITLE
Temperature and FPU related params.

### PR DESCRIPTION
**Pull request by mooskagh**
Introduce
--temp-cutoff-move (default: 0, i.e. switched off).
  Starting from the move specified in this parameter, the temperature is taken from --temp-endgame param.

--temp-value-cutoff
   When selecting move using temperature, only consider moves within X% (1% = 0.02 of Q value) from bestmove. Default is 100%, e.g. all moves.

FPU:
--fpu-strategy  Can be "reduction" (as before, value is taken from --fpu-reduction flag) or "absolute" (value is taken from --fpu-value flag)

### CODE REVIEW

File: src/mcts/search.cc [hunk 1]
```diff
@@ -447,8 +456,11 @@ void Search::EnsureBestMoveKnown() REQUIRES(nodes_mutex_)
...
   float temperature = params_.GetTemperature();
-  if (temperature && params_.GetTempDecayMoves()) {
-    int moves = played_history_.Last().GetGamePly() / 2;
+  const int cutoff_move = params_.GetTemperatureCutoffMove();
+  const int moves = played_history_.Last().GetGamePly() / 2;
+  if (cutoff_move && (moves + 1) >= cutoff_move) {
+    temperature = params_.GetTemperatureEndgame();
+  } else if (temperature && params_.GetTempDecayMoves()) {
     if (moves >= params_.GetTempDecayMoves()) {
       temperature = 0.0;
```
Comment (Tilps) [inline src/mcts/search.cc]: Why not use GetTemperatureEndgame here as well.

File: src/mcts/search.cc [hunk 2]
```diff
@@ -1038,11 +1052,11 @@ int SearchWorker::PrefetchIntoCache(Node* node, int budget) {
 
   float puct_mult = cpuct * std::sqrt(std::max(node->GetChildrenVisits(), 1u));
   // FPU reduction is not taken into account.
```
Comment (Tilps) [inline src/mcts/search.cc]: This comment is no longer correct?

File: src/mcts/search.cc (hunk 1 above)
Comment (mooskagh) [reply src/mcts/search.cc]: Decided not to do now. (unless you are willing to send a patch).

File: src/mcts/params.cc [hunk 3]
```diff
@@ -90,13 +102,27 @@ const OptionId SearchParams::kSmartPruningFactorId{
     "promising moves from being considered even earlier. Values less than 1 "
     "causes hopeless moves to still have some attention. When set to 0, smart "
     "pruning is deactivated."};
+const OptionId SearchParams::kFpuStrategyId{
+    "fpu-strategy", "FpuStrategy",
+    "How is an eval of unvisited node determined. \"reduction\" more subtracts "
```
Comment (borg323) [inline src/mcts/params.cc]: Delete `more`

**Comment by borg323** [comment]
Long term I think the best way is to make temperature take a list of temperatures and use a second option to set the breakpoints (maybe even reuse `--temp-cutoff-move`)

**Pull request merged**